packages/core/src/shorthand-parser/index.ts

```diff
@@ -292,3 +293,46 @@ export const boxShadow = (tokens: any, value: string) => {
       .join(', '),
   };
 };
+
+export const textDecoration = createPropertyParser((tokens: any, css: any, value: any) => {
+  if (matchString(value, /unset/)) {
+    // we are checking the existing value here because we want to retain the long hand property that was previously applied
+    /**
+     * For example:
+     * textDecorationColor: red
+     * textDecoration: unset
+     *
+     * The above should result in a final css rule like the following
+     * text-decoration-style: unset;
+     * text-decoration-line: unset;
+     * text-decoration-color: red
+     *
```

Comment:
Could you please walk me through this change as the css argument is an empty object that we append parsed longhand values onto so I'm not sure how this condition would be met. 

Would appreciate if you could write a test case for it as it would make it easier to understand

Reply:
You are right. I incorrectly assumed that the previous css property will exist on the css object.
So what I was trying to achieve by doing this was to mimic the browser behaviour.
On chrome when I was trying to do the following
```
text-decoration-color: red;
text-decoration: unset;
```

It was actually resulting in the following css:
```
text-decoration-color: red;
text-decoration-line: unset;
text-decoration-style: unset;
```

But since our css object is always initialized as an empty object, it seems like we can't do this.